spec/controllers/queue_items_controller_spec.rb

```diff
@@ -83,6 +83,161 @@
     end
   end
 
+  describe 'PATCH update' do
+    context 'with authenticated user' do
+
+      let(:current_user) { Fabricate(:user) }
+      let(:video1) { Fabricate(:video) }
+      let(:video2) { Fabricate(:video) }
+      let(:item1) { Fabricate(:queue_item, queue_position: -1) }
+      let(:item2) { Fabricate(:queue_item, queue_position: -1) }
+
+      before do
+        session[:user_id] = current_user.id
+
+        item1.update_attributes(user: current_user, video: video1)
+        item2.update_attributes(user: current_user, video: video2)
+
+        item1.rating = 1
+        item2.rating = 2
+      end
+
+      context 'updating queue positions' do
+        it 'redirects to my_queue_path for invalid positions' do
+          patch :update_queue, queue: [
+            {id: item1.id, position: 1.5, rating: 1}]
+
+          expect(response).to redirect_to my_queue_path
+        end
+
+        it 'sets the flash-error for invalid positions' do
+          patch :update_queue, queue: [
+            {id: item1.id, position: 1.5, rating: 1}
+          ]
+
+          expect(flash[:error]).not_to be_nil
```

Comment:
Another way to say `not_to be_nil` is `to `be_present`. Whenever possible, possible matchers are recommended. Negative matchers say "it should not be X", but they don't say what it SHOULD be.